Sources/Web3Core/KeystoreManager/BIP32HDNode.swift

```diff
@@ -90,212 +89,213 @@ public class HDNode {
 
     public init?(seed: Data) {
         guard seed.count >= 16 else { return nil }
-        // swiftlint:disable force_unwrapping
-        let hmacKey = "Bitcoin seed".data(using: .ascii)!
-        let hmac = HMAC(key: hmacKey.bytes, variant: HMAC.Variant.sha2(.sha512))
+
+        guard let hmacKey = "Bitcoin seed".data(using: .ascii) else { return nil }
+        let hmac: Authenticator = HMAC(key: hmacKey.bytes, variant: HMAC.Variant.sha2(.sha512))
+
         guard let entropy = try? hmac.authenticate(seed.bytes), entropy.count == 64 else { return nil }
         let I_L = entropy[0..<32]
         let I_R = entropy[32..<64]
         chaincode = Data(I_R)
         let privKeyCandidate = Data(I_L)
         guard SECP256K1.verifyPrivateKey(privateKey: privKeyCandidate) else { return nil }
         guard let pubKeyCandidate = SECP256K1.privateToPublic(privateKey: privKeyCandidate, compressed: true) else { return nil }
-        guard pubKeyCandidate.bytes[0] == 0x02 || pubKeyCandidate.bytes[0] == 0x03 else { return nil }
+        guard pubKeyCandidate.bytes.first == 0x02 || pubKeyCandidate.bytes.first == 0x03 else { return nil }
         publicKey = pubKeyCandidate
         privateKey = privKeyCandidate
         depth = 0x00
         childNumber = UInt32(0)
     }
 
     private static var curveOrder = BigUInt("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141", radix: 16)!
-    // swiftlint:enable force_unwrapping
     public static var defaultPath: String = "m/44'/60'/0'/0"
     public static var defaultPathPrefix: String = "m/44'/60'/0'"
     public static var defaultPathMetamask: String = "m/44'/60'/0'/0/0"
     public static var defaultPathMetamaskPrefix: String = "m/44'/60'/0'/0"
-    public static var hardenedIndexPrefix: UInt32 = (UInt32(1) << 31)
+    public static var hardenedIndexPrefix: UInt32 { Self.maxIterationIndex }
 }
 
 extension HDNode {
     public func derive(index: UInt32, derivePrivateKey: Bool, hardened: Bool = false) -> HDNode? {
         if derivePrivateKey {
-            return self.derivePrivateKey(index: index, hardened: hardened)
-        } else {
-            return derivePublicKey(index: index, hardened: hardened)
+            return deriveWithPrivateKey(index: index, hardened: hardened)
+        } else { // deriving only the public key
+            return deriveWithoutPrivateKey(index: index, hardened: hardened)
```

Comment:
What's the reason behind changing this? I think before it was more descriptive, so no comment would be needed.

Reply:
work towards cleaning up this massive block of code to make maintainability easier

Reply:
the maintainability of the function should be addressed